### PR TITLE
Add `unsafe` to two functions with safety invariants

### PR DESCRIPTION
**Pull request by eduardosm**

### CODE REVIEW

File: library/std/src/sys/pal/windows/thread.rs [hunk 1]
```diff
@@ -45,29 +45,29 @@ impl Thread {
...
+                }
             };
         };
     }
 
-    pub fn set_name_wide(name: &[u16]) {
-        unsafe {
-            c::SetThreadDescription(c::GetCurrentThread(), name.as_ptr());
-        };
+    // SAFETY: `name` must end with a zero value
```
Comment (ChrisDenton) [inline library/std/src/sys/pal/windows/thread.rs:68]: Could you make this a `\\\` doc comment with a `# Safety` heading?

**Review by ChrisDenton — COMMENTED** [review]
Thanks! Just one minor thing.

File: library/std/src/sys/pal/windows/thread.rs (hunk 1 above)
Comment (eduardosm) [reply library/std/src/sys/pal/windows/thread.rs]: Sure!

**Pull request merged**